app/models/concerns/host_info.rb

```diff
@@ -0,0 +1,24 @@
+module HostInfo
+  extend ActiveSupport::Concern
+
+  def info
+    info_hash = {}
+
+    self.class.info_providers.each do |provider_method|
```

Comment:
>Also I might be missing something here but is this about the ENC or is this about something else?

ENC is a Puppet term, but since we want to be configuration platform agnostic (we don't want to be locked to Puppet, but to be able to use other config tools like Ansible) we want to be able to expose our parameters to other systems, for example Ansible's Inventory.
Basically it means that `host.info` should describe the host in an abstract way - not something that is specific to puppet/chef/ansible or whatever, which it already does.